apps/passport-client/components/screens/ProveScreen/SemaphoreGroupProveScreen.tsx

```diff
@@ -35,15 +36,29 @@ export function SemaphoreGroupProveScreen({
   // Once that's done & user clicks Prove, create a zero-knowledge proof
   const [state] = useContext(DispatchContext);
   const [proving, setProving] = useState(false);
+
   const onProve = useCallback(async () => {
     setProving(true);
-    const serializedPCD = await prove(state.identity, group, req.args);
+    const args = await fillArgs(state.identity!, group, req.args);
 
-    // Redirect back to requester
-    window.location.href = `${req.returnUrl}?proof=${JSON.stringify(
-      serializedPCD
-    )}`;
-  }, [group, setProving, req.returnUrl, state.identity, req.args]);
+    if (req.options?.server === true) {
+      const serverReq: ProveRequest = {
+        pcdType: SemaphoreGroupPCDPackage.name,
+        args: args,
+      };
+      const pendingStampPCD = await requestStampPCD(serverReq);
+      window.location.href = `${req.returnUrl}?pendingStampPCD=${JSON.stringify(
+        pendingStampPCD
+      )}`;
+    } else {
+      const { prove, serialize } = SemaphoreGroupPCDPackage;
+      const pcd = await prove(args);
+      const serializedPCD = await serialize(pcd);
+      window.location.href = `${req.returnUrl}?proof=${JSON.stringify(
+        serializedPCD
+      )}`;
+    }
+  }, [group, req.returnUrl, state.identity, req.args, req.options?.server]);
```

Comment:
i think the way you've implemented it only the group prove screen has server-side proving enabled. I think we should probably extract this logic such that it is re-useable between this proving screen, the `SignatureProofScreen`, and the `GenericProofScreen`.